script.rb

```diff
@@ -0,0 +1,81 @@
+module Enumerable
+  def my_each(array)
+    index = 0
+    while index < array.length
+      yield(array[index])
+      index += 1
+    end
+    array
+  end
+
+  def my_each_with_index(array)
+    index = 0
+    while index < array.length
+      yield(array[index], index)
+      index += 1
+    end
+    array
+  end
+
+  def my_select(array)
+    new_arr = []
+    my_each(array) do |x|
+      new_arr.push(x) if yield(x) == true
+    end
+    new_arr
+  end
+
+  def my_all?(array)
+    p my_select(array) { |x| yield(x) }.length == array.length
+  end
+
+  def my_any?(array)
+    p my_select(array) { |x| yield(x) }.length.positive? ? true : false
+  end
+
+  def my_none?(array)
+    p my_select(array) { |x| yield(x) }.length.length.zero? ? true : false
+  end
+
+  def my_count(array, idx = 0)
+    index = idx
+
+    if block_given? == true
+      my_select(array) { |x| yield(x) }.length
+
+    else
+      array.length - index
+    end
+  end
+
+  def my_map_block(array)
+    new_arr = []
+    my_each(array) do |x|
+      new_arr.push(yield(x))
+    end
+    p new_arr
+  end
+
+  def my_map_proc(array, &block)
+    new_arr = []
+    my_each(array) do |x|
+      new_arr.push(block.call(x))
+    end
+    p new_arr
+  end
+
+  def my_inject(array, acc = 0)
+    if my_all?(array) { |x| x.class == String }
+      count = ''
+    elsif my_all?(array) { |x| x.class == Integer }
+      count = acc
+    else
+      return p 'no implicit conversion of Integer into String (TypeError)'
+    end
+
+    my_each(array) do |x|
+      count = yield(count, x)
+    end
+    p count
+  end
```

Comment:
- [ ] should have working `my_inject` **(covered by automated tests)** identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

####  my_inject
> Remember to test it not only for arrays but for ranges as well

- [ ] Raises a "LocalJumpError" when no block or argument is given

```ruby
[2,5,3].my_inject == [2,5,3].inject => LocalJumpError (no block given)
```
- Real inject retrieves a LocalJumpError (right answer)
![image](https://user-images.githubusercontent.com/52209067/93632451-96530080-f9aa-11ea-8c8c-7115a17aedcd.png)

- [ ] combines all elements of enum by applying a binary operation, specified by a block
```ruby
(1..5).my_inject(4) { |prod, n| prod * n } == (1..5).inject(4) { |prod, n| prod * n } => 480
```
- Real inject
![image](https://user-images.githubusercontent.com/52209067/77698409-2ffaa180-6f76-11ea-91d3-9778aa560411.png)

- [ ] When a symbol is provided as an argument, my_inject should combine the elements of the enum using that symbol as a named method.
```ruby
[2, 5, 6].my_inject(:+) == [2, 5, 6].inject(:+) => 13
```
- Real inject
![image](https://user-images.githubusercontent.com/52209067/77698534-6a643e80-6f76-11ea-9d0e-ebb0f02da179.png)